File: Announcements.Core/Entities/Announcement.cs
using System.ComponentModel.DataAnnotations;

namespace Announcements.Core.Entities
{
    public class Announcement
    {
        [Key]
        public string Title { get; set; }
        public string Desciption { get; set; }
        public DateTime DateAdded { get; set; }
    }
}

File: Announcements.WebAPI/ClientApp/AnnouncementsAngularApp/src/app/models/announcement.ts
export class Announcement {
  id: number;
  title: string | null;
  description: string | null;
  dateAdded: Date;

  constructor(id: number, title: string | null = null, description: string | null = null, dateAdded: Date) {
    this.id = id;
    this.title = title;
    this.description = description;

    this.dateAdded = dateAdded;
  }
}
